maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java

```diff
@@ -646,8 +657,33 @@ else if ( modelSource instanceof FileModelSource )
         {
             model.setPomFile( ( (FileModelSource) modelSource ).getFile() );
         }
-
         problems.setSource( model );
+
+        modelValidator.validateFileModel( model, request, problems );
+        request.setFileModel( model );
+        
+        if ( Features.buildConsumer().isActive() && pomFile != null )
+        {
+            try
```

Comment:
Would it be possible to refactor this addition (and others) to the `readModel` method into separate methods? The fact that we now need to `@SuppressWarnings` about method length is a sign that the method is, well, lengthy ;-).

Reply:
Building the model is matter of a lot of steps. I gave it a try, but it actually made the code harder to understand, because some clear if/else or foreach statements make more sense within this context.
One might give it a try, but not as part of this ticket.